src/rgw/rgw_sal_motr.cc

```diff
@@ -1379,6 +1379,10 @@ int MotrObject::MotrReadOp::prepare(optional_yield y, const DoutPrefixProvider*
     }
   }
 
+  // Skip opening an empty object.
```

Comment:
MotrObject::read_mobj() does not check it the object is opened or not. It might result in error.
We should also check the object size inside MotrObject::MotrReadOp::iterate or directly handle it in RGWGetObj::execute() function.

Reply:
It did not error out... but got your point. let me check where to add this.

Reply:
Hi Just check the reason, 
```c++
  if (!get_data || ofs > end) {
    send_response_data(bl, 0, 0);
    return;
  }
```
since the off = 0 & end = -1, we go to send_response_data directly without even hitting the read_op->iterate method.
so it's already handled in the execute() method 👍

Reply:
tested it. worked fine.